src/odemis/driver/actuator.py

```diff
@@ -1366,3 +1366,510 @@ def terminate(self):
+                                  "Value should be in range -%s/2 and +%s/2." % (value, key, self._cycle, self._cycle))
+                raise ValueError("value %s for metadata entry %s is not allowed." % (value, key))
+
+    def _on_referenced(self, future):
+        try:
+            future.result()
+        except Exception as e:
+            self._child.stop({self._caxis})  # prevent any move queued
+            self.state._set_value(e, force_write=True)
+            logging.exception(e)
+
+    def _updatePosition(self, _=None):
+        """
+        update the position VA
+        """
+        p = self._child.position.value[self._caxis] - self._metadata[model.MD_POS_COR]
+        if self._cycle is not None:
+            p %= self._cycle
+        self._position[self._axis] = p
+        real_pos = self._position[self._axis]
+        pos = {self._axis: real_pos}
+        logging.debug("reporting position %s", pos)
+        self.position._set_value(pos, force_write=True)
+
+    def _updateReferenced(self, _=None):
+        """
+        update the referenced VA
+        """
+        self._referenced[self._axis] = self._child.referenced.value[self._caxis]
+        # .referenced is copied to detect changes to it on next update
+        self.referenced._set_value(self._referenced.copy(), force_write=True)
+
+    @isasync
+    def moveRel(self, shift):
+        if not shift:
+            return model.InstantaneousFuture()
+        self._checkMoveRel(shift)
+        raise NotImplementedError("Relative move on fixed positions axis not supported")
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        Move the actuator to the defined position in m for each axis given.
+        pos dict(string-> float): name of the axis and position in m
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        self._checkMoveAbs(pos)
+        pos = self._applyInversion(pos)
+        f = CancellableFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, pos)
+        f.task_canceller = self._cancelMovement
+
+        return f
+
+    def _cancelMovement(self, future):
+        if self._child_future:
+            return self._child_future.cancel()
+        return False
+
+    def _doMoveAbs(self, pos):
+        axis, position = pos.items()[0]
+        # correct distance for physical offset due to mounting
+        position = position + self._metadata[model.MD_POS_COR]
+        logging.debug("Moving axis %s (-> %s) to %g", self._axis, self._caxis, position)
+
+        # do referencing after i=5 moves or when actuator has accumulated 2pi steps
+        # --> do correct for accumulated errors
+        if self._move_num_total == 5 or abs(self._move_sum) >= self._cycle:
+            # to get rid of accumulated error
+            self._move_sum = 0
+            self._move_num_total = 0
+
+            # move to the reference switch (chose 5% of 2pi)
+            move_to_ref = (2 * numpy.pi) * 0.05
```

Comment:
mhm true, but I am not sure if it makes sense to have self._offset. One revolution is always 2pi, however cycle can be also 4pi. For move to referencing switch I think it would be enough to hardcode 5% of 2pi and remove self._offset....

Reply:
Done.